src/routes/github/setup/github-setup-get.ts

```diff
@@ -14,32 +14,25 @@ import { Installation } from "../../../models";
 */
 export const GithubSetupGet = async (req: Request, res: Response): Promise<void> => {
 	req.log.info("Received get github setup page request");
-	const { jiraHost, github, client } = res.locals;
-	const { data: { installations } } = await github.apps.listInstallationsForAuthenticatedUser();
+	const { jiraHost, client } = res.locals;
+	const installationId = Number(req.query.installation_id);
 	const { data: info } = await client.apps.getAuthenticated();
-	const installationId = req.query.installation_id;
-	const installation = installations.find((item) => item.id === Number(installationId));
-
-	if (!installation) {
-		throw new Error(`Error retrieving installation:${installationId}. You do not permissions to access this org or it does not exist.`);
-	}
-
-	let redirectUrl = getJiraMarketplaceUrl(jiraHost);
+	const githubInstallation = await client.apps.getInstallation({ installation_id: installationId })
+		.catch(() => {
+			// if we cannot get github installation, try to log as much as possible to help debug
+			req.log.warn({ appInfo: info, jiraHost, installationId }, "Cannot retrieve Github Installation from API");
+		});
```

Comment:
Using app based API instead of user based API to get the installation directly based on ID.  Also making it optional as we only use it to show the user the current github org name/avatar that they've just connected.

Reply:
Actually, now that I think about it, if we know the `jiraHost` at this point, we should be able to allow the user to connect to their instance right away.